Language/Expressions/Property_Extraction/Instance_Method_Closurization/method_closurization_named_params_A06_t02.dart

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Let `o` be an object, and let `u` be a fresh final variable bound
+/// to o. The closurization of method `f` on object `o` is defined to be
+/// equivalent to:
+/// ```
+/// - <X1 extends B′1, ..., Xs extends B′s>
+///     (T1 p1, ..., Tn pn, {Tn+1 pn+1 = d1, ..., Tn+k pn+k = dk}) =>
+///       u.m<X1, ..., Xs>(p1, ..., pn, pn+1: pn+1, ..., pn+k: pn+k);
+///  ```
+///  where `f` is an instance method named `m` which has type parameter
+///  declarations `X1 extends B1, ..., Xs extends Bs`, required parameters
+///  `p1, ..., pn`, and named parameters `pn+1, ..., pn+k` with defaults
+///  `d1, ..., dk`, using `null` for parameters whose default value is not
+///  specified.
+///  ...
+///  There is one way in which the function object yielded by the instance
+///  method closurization differs from the function object obtained by function
+///  closurization on the above mentioned function literal: Assume that `o1` and
+///  `o2` are objects, `m` is an identifier, and `c1` and `c2` are function
+///  objects obtained by closurization of `m` on `o1` respectively `o2`. Then
+///  `c1 == c2` evaluates to `true` if and only if `o1` and `o2` is the same
+///  object.
+///
+/// @description Check that if `o1` and `o2` are objects, `m` is an identifier,
+/// and `c1` and `c2` are function objects obtained by closurization of `m` on
+/// `o1` respectively `o2`, then `c1 == c2` evaluates to `false` if `o1` and
+/// `o2` are extensions or extension types.
+/// @author sgrekhov22@gmail.com
+/// @issue 60065
+
+import '../../../../Utils/expect.dart';
+
+class A {}
+extension Ext on A {
+  num m(int r1, {String p1 = ""}) => r1;
+}
+
+extension type ET(int _) {
+  num m(int r1, {p1 = ""}) => r1;
+}
+
+main() {
+  A a = A();
+  final fa1 = a.m;
+  final fa2 = a.m;
+  Expect.notEquals(fa1, fa2);
```

Comment:
Note that the spec wording says that the two tear-offs are equal if and only if they are identical. But it doesn't say that they can't be identical.

In this particular case we actually need to preserve the permission to optimize the extension methods: If an extension method doesn't actually depend on the receiver (as in this example: it just returns an argument and never uses `this`) then it must be OK for an implementation to compile this extension method to, for example, a simple top-level function like `num compiled_m(int r1, {String p1 = ""}) => r1;` and then return that (simple and constant) function object when the corresponding extension method is torn off.

In this case we'll have `identical(fa1, fa2)` (and hence also `fa1 == fa2`).

So we have to weaken the test to say that it's OK for `fa1` and `fa2` to be unequal, and it's also OK for them to be equal (and even identical, if anyone asks ;-).

As usual, evaluation of `fa1 == fa2` shouldn't throw, so we shouldn't just skip the test entirely.

So perhaps this would do?:

```suggestion
  Expect.isTrue(fa1 == fa2 || fa1 != fa2);
```

Reply:
Thank you. Makes sense. I simply deleted this test. Expect.isTrue(fa1 == fa2 || fa1 != fa2); will hardly ever fail and will always be true. Therefore, this test does nothing except confirm that extension and extension type methods can be torn off. We have lots of other tests (including ones in this PR) that check this.

Reply:
Do we have a test that verifies that the operator `==` on these function objects isn't buggy and throws?

Reply:
Please clarify, how `==` may throw on such function objects?

Reply:
That would be a bug, but bugs is the main reason why we have tests.